include/Beman/Optional26/detail/iterator.hpp

```diff
@@ -0,0 +1,73 @@
+// include/Beman/Optional26/detail/iterator.hpp -*-C++-*-
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef BEMAN_OPTIONAL26_DETAIL_ITERATOR_HPP
+#define BEMAN_OPTIONAL26_DETAIL_ITERATOR_HPP
+
+#include <Beman/Optional26/detail/stl_interfaces/iterator_interface.hpp>
+
+#include <concepts>
+#include <iterator>
+
+namespace beman::optional::detail {
+
+// Forward declaration.
+template <class T, class Container>
+struct contiguous_iterator;
+
+// Base class for contiguous iterator types with Boost stl_interfaces library.
+// Current implementation based on P2727R4: std::iterator_interface.
+template <class T, class Container>
+using base_contiguous_iterator = stl_interfaces::iterator_interface<
+#if !BEMAN_OPTIONAL26_DETAIL_STL_INTERFACES_USE_DEDUCED_THIS
+    contiguous_iterator<T, Container>, // Required for P2727R4 to work with C++20/C++23. TODO: Do more experiments.
+#endif
+    std::contiguous_iterator_tag,
+    T>;
+
+// This is a minimal contiguous iterator.  It uses stl_interfaces library from Boost
+// (current implementation based on https://wg21.link/P2727R4).
+//
+// TODO: Change this to use the stl_interfaces library from Beman if/when available.
+//
+// @tparam T - The type of the elements the iterator points to.
+// @tparam Container - The type of the container the iterator points to. This parameter exists solely so that different
+// containers using this template can instantiate different types, even if the T parameter is the same.
+template <class T, class Container>
+struct contiguous_iterator : public base_contiguous_iterator<T, Container> {
+    // Alias for the base class.
+    using base_type = base_contiguous_iterator<T, Container>;
+    // Alias for types from the base class.
+    using typename base_type::difference_type;
```

Comment:
It looks like you need difference_type and pointer, but I don't think you need the rest.

Reply:
right, also need reference. removed unused aliases